### ISSUE TITLE
Goals

### ISSUE THREAD

**Issue by tkell** [algorithmic-music-exploration/amen-server#1]
Basically, we want to recreate something very much like The Echo Nest's `track/upload` endpoint.  This means that:

- The user should be able to `POST` audio to the server and get a JSON blob of analysis back.
- The user may be able to access the audio on some sort of S3-like system so they can use it in JavaScript projects.
- The analysis may or may not emulate Echo Nest JSON. 

It would be super-ideal if things like The Infinite Jukebox could just use our endpoint.  May not be possible tho.

**Comment by alnesbit** [comment]
I'm curious about straightening out a few concepts and ensuring that I understand them.

> The user may be able to access the audio on some sort of S3-like system so they can use it in JavaScript projects.

What is "the audio"? Are you referring, for example, to the audio samples generated by librosa? So, conceptually, instead of writing the audio samples in an `amen.audio.Audio` object to a local file, they would get written to an arbitrary location; in this case, S3?

Less important for now, but keeping in mind for future:
* Regarding S3, is there anything that necessarily ties us to using this? Would there be any advantage in writing audio to some other block storage in the cloud, locally, on the local network, etc? The OpenStack API, for example, is a potential alternative or option. In any case, S3 is really fine. I'm not necessarily advocating for overarchitecting, generalization, or alternative APIs. I only would like to clarify the focus on S3.
* Regarding "JavaScript projects", is there any reason to refer specifically to JS, or is this just an example?

> It would be super-ideal if things like The Infinite Jukebox could just use our endpoint. May not be possible tho.

In other words, would it be ideal if amen-server could be a fully compatible, drop-in replacement?

**Comment by tkell** [comment]
Hey hey!

- "The audio" means the uploaded and anaylzed raw audio file, not analysis an `Audio` object.  
- No need to use S3, in particular - could be OpenStack, could be Google Cloud Storage, etc
- JS projects are just an example, yep.  In general, because Amen does local analysis, if you're doing something with a script, you can just use Amen.  If you want to allow user uploads, or do something with a web or mobile UI, you can use Amen Server
- I would love love love if Amen Server returned JSON that looked like Echo Nest JSON, for sure.
